tests/databricks_ai_bridge/test_genie.py

```diff
@@ -232,6 +234,81 @@ def test_parse_query_result_trims_large_data():
         assert _count_tokens(result) <= 100
 
 
+def markdown_to_dataframe(markdown_str: str) -> pd.DataFrame:
+    lines = markdown_str.strip().splitlines()
+
+    # Remove Markdown separator row (2nd line)
+    lines = [line.strip().strip("|") for i, line in enumerate(lines) if i != 1]
+
+    # Re-join cleaned lines and parse
+    cleaned_markdown = "\n".join(lines)
+    df = pd.read_csv(StringIO(cleaned_markdown), sep="|")
+
+    # Strip whitespace from column names and values
+    df.columns = [col.strip() for col in df.columns]
+    df = df.applymap(lambda x: x.strip() if isinstance(x, str) else x)
+
+    # Drop the first column
+    df = df.drop(columns=[df.columns[0]])
+
+    return df
+
+
+@pytest.mark.parametrize("max_tokens", [100, 1000, 2000, 8000, 10000, 15000, 19000, 100000])
```

Comment:
If we're doing a binary search, can we test some of the edge cases? e.g. 1 row is too large (MAX_TOKENS=1), odd numbers of rows exceeding the limit, etc